src/blas/backends/syclblas/syclblas_level2.cxx

```diff
@@ -147,7 +147,11 @@ void spmv(sycl::queue &queue, oneapi::mkl::uplo upper_lower, std::int64_t n, rea
 
 void spr(sycl::queue &queue, oneapi::mkl::uplo upper_lower, std::int64_t n, real_t alpha,
          sycl::buffer<real_t, 1> &x, std::int64_t incx, sycl::buffer<real_t, 1> &a) {
-    throw unimplemented("blas", "spr", "");
+    oneapi::mkl::uplo Uplo = (is_column_major() && upper_lower == oneapi::mkl::uplo::upper) ||
+                                     (!is_column_major() && upper_lower == oneapi::mkl::uplo::lower)
+                                 ? oneapi::mkl::uplo::upper
+                                 : oneapi::mkl::uplo::lower;
+    CALL_SYCLBLAS_FN(::blas::_spr, queue, Uplo, n, alpha, x, incx, a);
```

Comment:
@hjabird  Thanks for mentioning it, currently SYCL-BLAS has no Row-major implementation for any of the functions we have. SYCL-BLAS is column major only for all cases. Once the row-major integration is added, it should add be integrated for oneMKL Row-major. Hence, all the SYCL-BLAS integration for oneMKL should only include the column-major for now and throw un-implemented for the row-major.

Reply:
Hi @hjabird, Thanks for pointing this out. Like Mehdi said, SYCL-BLAS only supports `col-major` layout at the moment. However, `spr` operator has this unique property that it can work the same for both `col-major` and `row-major` layouts by carefully selecting the `upper` and `lower` triangular matrix layouts. That's is the reason I put this check here to select the appropriate value for `Uplo` variable based on whether the layout is `col-major` or `row-major`.

Reply:
Hi @hjabird, your explanation above was correct, I was hoping that the `RowMajor` tests would run with the changes I made for `spr` but it seems like it requires some other changes as well. I have removed the conditional statement for now and will hopefully address this when we add `row-major` support for SYCL-BLAS backend.